src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1032,105 +1087,98 @@ private void RemoveWorseMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMe
             // then we need to do no more work; we know it cannot win. But it is also possible that
             // it is not worse than anything but not better than everything. 
 
+            if (SingleValidResult(results))
+            {
+                return;
+            }
+
+            // See if we have a winner, otherwise we might need to perform additional analysis
+            // in order to improve diagnostics
+            int bestIndex = GetTheBestCandidateIndex(results, arguments, ref useSiteDiagnostics);
+            if (bestIndex != -1)
+            {
+                var bestMember = results[bestIndex].Member;
+                // Mark all other candidates as worse
+                for (int index = 0; index < results.Count; index++)
+                {
+                    if (results[index].IsValid && index != bestIndex && results[index].Member != bestMember)
```

Comment:
> && results[index].Member != bestMember [](start = 68, length = 39)

I don't think it is appropriate to do this filtering here. Theoretically the member could be marked with notBetterThanEverything by the old algorithm, therefore it would be marked as Worse. Please consider my suggestions for GetTheBestCandidateIndex instead.